xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -249,6 +249,8 @@ def extract_var(self, a: ParsingState) -> ConstraintVariableType:
             assert types is not None
             if isinstance(types, Attribute):
                 types = (types,)
+            else:
```

Comment:
This looks like it introduces a copy rather than getting rid of any. Am I missing something?